tests/components/icloud/test_config_flow.py

```diff
@@ -0,0 +1,205 @@
+"""Tests for the iCloud config flow."""
+import pytest
+from unittest.mock import patch
+from pyicloud.exceptions import PyiCloudFailedLoginException
+
+from homeassistant import data_entry_flow
+from homeassistant.components.icloud import config_flow
+from homeassistant.components.icloud.config_flow import CONF_TRUSTED_DEVICE
+from homeassistant.components.icloud.const import (
+    DOMAIN,
+    CONF_ACCOUNT_NAME,
+    CONF_GPS_ACCURACY_THRESHOLD,
+    CONF_MAX_INTERVAL,
+    DEFAULT_GPS_ACCURACY_THRESHOLD,
+    DEFAULT_MAX_INTERVAL,
+)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+USERNAME = "username@me.com"
+PASSWORD = "password"
+ACCOUNT_NAME = "Account name 1 2 3"
+ACCOUNT_NAME_FROM_USERNAME = "username"
+MAX_INTERVAL = 15
+GPS_ACCURACY_THRESHOLD = 250
+
+
+@pytest.fixture(name="init")
+def mock_controller_authenticate():
+    """Mock a successful authenticate."""
+    with patch("pyicloud.PyiCloudService", return_value=None):
```

Comment:
The return value should not be None, when successful. Just don't specify anything to set a magic mock as return value.